Add the ability to output the computed values using the template sub-command

jamoflaw opened the pull request:
**What this PR does / why we need it**:

This adds the ability to template the resolved values files (all the values including the chart defaults). This is equivalent to the output seen in the `--dry-run` `COMPUTED VALUES:` section.

The default behaviour of template is untouched and only changed when you use the `--computed-values` flag, which will output 
 the computed values _instead_ of the fully rendered manifest.

**example:**
```
# Display the computed values of the chart in its default state
helm template ./chart/name --computed-values
# Display the computed values of the chart after overrides have been applied
helm template ./chart/name --computed-values -f overrides.yaml
```

will output the values _after_ they have been resolved by the helm parsers.

This is useful in simple CI situations where a particular value is expected especially when traversing multiple environments, it is also useful as an input for other parsers or templating tools to provide customised output based on the values passed into the chart.

- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

jamoflaw committed:
Add the ability to output the computed values using the template sub-command

The default behaviour of template is untouched and only changed when you use
the --computed-values flag which will output the computed values _instead_ of
the fully rendered manifest.

example:
helm template ./chart/name --computed-values

will output the values _after_ they have been resolved by the helm parsers.

Signed-off-by: James Payne <jamoflaw@gmail.com>

cndoit18 commented:
PRs must be linked to an existing issue. If none exists, may you please open one explaining the issue and expected outcome?
Please also write tests to test and verify this PR fixes your issue. Thanks.

cndoit18 reviewed (commented):
Please add the unit test.

jamoflaw commented:
Ref issue #9883

jamoflaw committed:
Add unit-tests to cover PR #9873

Tests expect the same YAML output as would be provided by the --debug --dry-run computed values

Signed-off-by: James Payne <jamoflaw@gmail.com>

jamoflaw commented:
Have added some unit tests to cover the output - in all cases the output is the same as the output from `install` with `--debug --dry-run`

Run the tests locally and passes without issue
